From c41f1c6b8cb136037e1cc89cdca671c9a53847bd Mon Sep 17 00:00:00 2001
From: Mahesh Mahadevan <r9aadq@freescale.com>
Date: Wed, 2 Nov 2011 11:12:23 -0500
Subject: [PATCH 381/678] ENGR00161296 Update the MX6 Sabre-lite environment
 option in the config file

Allow boot to either SD card through 6q_bootscript.
Define clearenv command to restore factory defaults
Add upgradeu command to upgrade u-boot if required

Signed-off-by: Mahesh Mahadevan <r9aadq@freescale.com>
---
 include/configs/mx6q_sabrelite.h |   27 ++++++++++++++++++++-------
 1 files changed, 20 insertions(+), 7 deletions(-)

diff --git a/include/configs/mx6q_sabrelite.h b/include/configs/mx6q_sabrelite.h
index 3285a9c..b346d3c 100644
--- a/include/configs/mx6q_sabrelite.h
+++ b/include/configs/mx6q_sabrelite.h
@@ -117,17 +117,30 @@
 		"uboot=u-boot.bin\0"			\
 		"kernel=uImage\0"				\
 		"nfsroot=/opt/eldk/arm\0"				\
-		"bootargs_base=setenv bootargs console=ttymxc1,115200\0"\
+		"bootargs=console=ttymxc1,115200\0" 			\
+		"bootcmd_base=\0" \
 		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs "\
 			"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"\
-		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
+		"bootcmd_net=run bootcmd_base bootargs_nfs; "	       	\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
-		"bootargs_mmc=setenv bootargs ${bootargs} "     \
-			"root=/dev/mmcblk0p1 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
-		"mmc dev 1; "	\
-		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
+		"bootcmd_mmc=run bootcmd_base ;" 			\
+			"for disk in 0 1 ; do mmc dev ${disk} ;" 	\
+				"for fs in fat ext2 ; do " 		\
+					"${fs}load mmc ${disk}:1 10008000 " \
+						"/6q_bootscript && " 	\
+					"source 10008000 ; " 		\
+				"done ; " 				\
+			"done\0" 					\
 		"bootcmd=run bootcmd_mmc\0"                             \
+		"clearenv=sf probe 1 && sf erase 0xc0000 0x2000 && "	\
+			"echo restored environment to factory default\0"\
+		"upgradeu=for disk in 0 1 ; do mmc dev ${disk} ;" 	\
+				"for fs in fat ext2 ; do " 		\
+					"${fs}load mmc ${disk}:1 10008000 " \
+						"/6q_upgrade && " 	\
+					"source 10008000 ; " 		\
+				"done ; " 				\
+			"done\0" 					\
 
 
 #define CONFIG_ARP_TIMEOUT	200UL
-- 
1.7.7.4

